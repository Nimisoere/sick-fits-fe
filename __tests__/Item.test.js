import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const fakeItem = {
    id: '1234',
    title: 'A cool item',
    price: 4000,
    description: 'Test item',
    image: 'test.jpg',
    largeImage: 'test.jpg'
};

describe('<Item />', () => {
    it('renders and matches snapshot', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('renders the price and title properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem } />);
        const PriceTag = wrapper.find('PriceTag');
        const price = PriceTag.children().text();
        expect(price).toBe('$40');

        const title = wrapper.find('Title a').text();
        expect(title).toBe(fakeItem.title);
    });
    it('renders the image properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem } />);
        const img = wrapper.find('img');
        const {src, alt} = img.props();
        expect(src).toBe(fakeItem.image);
        expect(alt).toBe(fakeItem.title);
    });
    it('renders the buttons properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem } />);
        const buttonList = wrapper.find('.buttonList').children();
        expect(buttonList).toHaveLength(3);
        expect(buttonList.find('Link')).toHaveLength(1);
        expect(buttonList.find('Link').exists()).toBe(true);
        expect(buttonList.find('AddToCart').exists()).toBe(true);
        expect(buttonList.find('DeleteItem').exists()).toBe(true);

    });
});