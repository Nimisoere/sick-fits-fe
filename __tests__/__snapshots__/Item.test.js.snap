// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="test.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "1234",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    Test item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "1234",
          },
        }
      }
    >
      <a>
        Edit 
        ‚úè
      </a>
    </Link>
    <AddToCart
      id="1234"
    />
    <DeleteItem
      id="1234"
    >
      Delete item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
